src/viam/sdk/common/proto_value.hpp

```diff
@@ -47,8 +47,14 @@ struct all_moves_noexcept
 /// A ProtoValue can be nullptr, bool, int, double, std::string, or, recursively, a vector or
 /// string-map of ProtoValue.
 /// This type is used at API/ABI boundaries for interfacing with grpc/proto code.
+/// @remark The (special) member function templates below all operate on a closed subset of types,
+/// so we provide explicit instantiations for all valid template types. See below the class
+/// definition.
 class ProtoValue {
    public:
+    /// @brief Type discriminator constants for possible values stored in a ProtoValue.
+    enum Kind { null, bool_, int_, double_, string, list, struct_ };
```

Comment:
- We usually prefix constants with `k_` (google style thing, but one I like), so `k_null`, `k_bool` will all work I think.
- I think maybe you should give these explicit numeric values since they form part of the interface.
- Maybe make the order the same order as the struct proto does: https://github.com/protocolbuffers/protobuf/blob/main/src/google/protobuf/struct.proto#L66-L76. Throw `int` at the end since we expect to remove it.

Reply:
Sounds good, going to do all these except keep the same order we have (but with `int` thrown at the end still). The reason for the proto sequencing escapes me, and I think it's easier mnemonically to write switch statements and sequence the methods if it goes from least to most data-rich types which is also what you see, eg, in JSON library APIs usually